Support K2

Currently, trying to build with KSP2 / K2 fails with: 

```
Caused by: org.gradle.workers.internal.DefaultWorkerExecutor$WorkExecutionException: A failure occurred while executing com.google.devtools.ksp.gradle.KspAAWorkerAction
	<junk>
Caused by: java.lang.NoClassDefFoundError: javaslang/collection/HashMap
	at org.jetbrains.kotlin.load.java.structure.impl.classFiles.ClassifierResolutionContext.<init>(ClassifierResolutionContext.kt:40)
	at com.google.devtools.ksp.impl.symbol.kotlin.UtilKt$getDefaultValue$1$2$visitMethod$1.visitAnnotationDefault(util.kt:435)
	at org.jetbrains.org.objectweb.asm.ClassReader.readMethod(ClassReader.java:1364)
	at org.jetbrains.org.objectweb.asm.ClassReader.accept(ClassReader.java:712)
	at org.jetbrains.org.objectweb.asm.ClassReader.accept(ClassReader.java:395)
	at com.google.devtools.ksp.impl.symbol.kotlin.UtilKt.getDefaultValue(util.kt:422)
	at com.google.devtools.ksp.impl.symbol.kotlin.KSAnnotationImpl$defaultArguments$2.invoke(KSAnnotationImpl.kt:77)
	at com.google.devtools.ksp.impl.symbol.kotlin.KSAnnotationImpl$defaultArguments$2.invoke(KSAnnotationImpl.kt:60)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.google.devtools.ksp.impl.symbol.kotlin.KSAnnotationImpl.getDefaultArguments(KSAnnotationImpl.kt:60)
	at com.google.devtools.ksp.impl.symbol.kotlin.KSAnnotationImpl$arguments$2.invoke(KSAnnotationImpl.kt:54)
	at com.google.devtools.ksp.impl.symbol.kotlin.KSAnnotationImpl$arguments$2.invoke(KSAnnotationImpl.kt:51)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.google.devtools.ksp.impl.symbol.kotlin.KSAnnotationImpl.getArguments(KSAnnotationImpl.kt:51)
	at com.ramcosta.composedestinations.ksp.processors.KspToCodeGenDestinationsMapper.toDestination(KspToCodeGenDestinationsMapper.kt:557)
	at com.ramcosta.composedestinations.ksp.processors.KspToCodeGenDestinationsMapper.access$toDestination(KspToCodeGenDestinationsMapper.kt:67)
	at com.ramcosta.composedestinations.ksp.processors.KspToCodeGenDestinationsMapper$map$1.invoke(KspToCodeGenDestinationsMapper.kt:78)
	at com.ramcosta.composedestinations.ksp.processors.KspToCodeGenDestinationsMapper$map$1.invoke(KspToCodeGenDestinationsMapper.kt:78)
	at kotlin.sequences.TransformingSequence$iterator$1.next(Sequences.kt:210)
	at kotlin.sequences.SequencesKt___SequencesKt.toList(_Sequences.kt:811)
	at com.ramcosta.composedestinations.ksp.processors.KspToCodeGenDestinationsMapper.map(KspToCodeGenDestinationsMapper.kt:78)
	at com.ramcosta.composedestinations.ksp.processors.Processor.process(Processor.kt:55)
	at com.google.devtools.ksp.impl.KotlinSymbolProcessing.execute(KotlinSymbolProcessing.kt:464)
	<junk>
	... 2 more
Caused by: java.lang.ClassNotFoundException: javaslang.collection.HashMap
	... 56 more
```

`gradle.properties`:
```properties
kotlin.experimental.tryK2=true
android.lint.useK2Uast=true
ksp.useKSP2=true
```

--- a/compose-destinations-ksp/src/main/kotlin/com/ramcosta/composedestinations/ksp/commons/Utils.kt
+++ b/compose-destinations-ksp/src/main/kotlin/com/ramcosta/composedestinations/ksp/commons/Utils.kt
@@ -249,7 +249,14 @@ fun KSValueParameter.toParameter(
     )
 }
 
-fun KSType.toGenVisibility(): Visibility {
+fun Any.toGenVisibility(): Visibility {
+    // Enum entries in annotation arguments are evaluated as KSClassDeclaration instead of KSType in KSP2
+    // https://github.com/google/ksp/blob/main/docs/ksp2api.md#evaluation-of-enum-entries-in-annotation-arguments
+    val declaration = when (this) {
+        is KSType -> declaration
+        is KSClassDeclaration -> this
+        else -> error("Unexpected type for 'visibility' param $this")
+    }
     return when (val visibility = declaration.simpleName.asString()) {
         "PUBLIC" -> Visibility.PUBLIC
         "INTERNAL" -> Visibility.INTERNAL

--- a/compose-destinations-ksp/src/main/kotlin/com/ramcosta/composedestinations/ksp/processors/KspToCodeGenDestinationsMapper.kt
+++ b/compose-destinations-ksp/src/main/kotlin/com/ramcosta/composedestinations/ksp/processors/KspToCodeGenDestinationsMapper.kt
@@ -204,7 +204,7 @@ internal class KspToCodeGenDestinationsMapper(
     }
 
     private fun KSAnnotation.getDestinationVisibility(): Visibility? {
-        return findArgumentValue<KSType>("visibility")?.toGenVisibility()
+        return findArgumentValue<Any>("visibility")?.toGenVisibility()
     }
 
     private val activityType by lazy {

--- a/compose-destinations-ksp/src/main/kotlin/com/ramcosta/composedestinations/ksp/processors/KspToCodeGenNavGraphsMapper.kt
+++ b/compose-destinations-ksp/src/main/kotlin/com/ramcosta/composedestinations/ksp/processors/KspToCodeGenNavGraphsMapper.kt
@@ -71,7 +71,7 @@ internal class KspToCodeGenNavGraphsMapper(
         val navGraphAnnotationNameArg = navGraphAnnotation
             .findArgumentValue<String>(DESTINATION_ANNOTATION_ROUTE_ARGUMENT)
         val navGraphVisibility = navGraphAnnotation
-            .findArgumentValue<KSType>("visibility")!!
+            .findArgumentValue<Any>("visibility")!!
             .toGenVisibility()
         val navGraphDefaultTransitions = navGraphAnnotation
             .findArgumentValue<KSType>("defaultTransitions")

